Fix Travis CI CVE Report (dom4j)

[Travis CI is failing with](https://travis-ci.org/dropwizard/dropwizard/jobs/452517897#L4127)

```
[ERROR] One or more dependencies were identified with vulnerabilities: 
[ERROR] 
[ERROR] dom4j-1.6.1.jar: CVE-2018-1000632
```

I'm away from a dev machine, so I can't check to see if there is an update for this transitive dependency, or if it is better to ignore it. Either way, it is bothersome to see PRs fail

--- a/dropwizard-bom/pom.xml
+++ b/dropwizard-bom/pom.xml
@@ -149,7 +149,7 @@
             <dependency>
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-core</artifactId>
-                <version>5.3.6.Final</version>
+                <version>5.3.7.Final</version>
                 <exclusions>
                     <exclusion>
                         <groupId>org.jboss.logging</groupId>

